funding/manager.go

```diff
@@ -619,42 +637,20 @@ func (f *Manager) start() error {
 			if chanType.IsSingleFunder() && chanType.HasFundingTx() &&
 				channel.IsInitiator {
 
-				var fundingTxBuf bytes.Buffer
-				err := channel.FundingTxn.Serialize(&fundingTxBuf)
-				if err != nil {
-					log.Errorf("Unable to serialize "+
-						"funding transaction %v: %v",
-						channel.FundingTxn.TxHash(), err)
-
-					// Clear the buffer of any bytes that
-					// were written before the serialization
-					// error to prevent logging an
-					// incomplete transaction.
-					fundingTxBuf.Reset()
-				}
-
-				log.Debugf("Rebroadcasting funding tx for "+
-					"ChannelPoint(%v): %x",
-					channel.FundingOutpoint,
-					fundingTxBuf.Bytes())
-
-				// Set a nil short channel ID at this stage
-				// because we do not know it until our funding
-				// tx confirms.
-				label := labels.MakeLabel(
-					labels.LabelTypeChannelOpen, nil,
-				)
-
-				err = f.cfg.PublishTransaction(
-					channel.FundingTxn, label,
-				)
-				if err != nil {
-					log.Errorf("Unable to rebroadcast "+
-						"funding tx %x for "+
-						"ChannelPoint(%v): %v",
-						fundingTxBuf.Bytes(),
-						channel.FundingOutpoint, err)
-				}
+				f.rebroadcastFundingTx(channel)
+			}
+		} else {
+			// Zero-conf channels with OtherShortChannelID unset
+			// should have their funding tx rebroadcast. This is an
+			// option_scid_alias channel even though
+			// IsOptionScidAlias returns false. This is because the
+			// OtherShortChanID does not yet exist.
+			cType := channel.ChanType
```

Comment:
Can avoid indention by making this an `else if` or `case` block (folded into the above).

Reply:
I don't understand why we wouldn't want to _always_ re-broadcast the funding transaction though (if we have the funding transaction).

Reply:
I don't think it would make sense to rebroadcast for confirmed channels